packages/typespec-client-generator-core/src/internal-utils.ts

```diff
@@ -619,3 +631,59 @@ export function findRootSourceProperty(property: ModelProperty): ModelProperty {
   }
   return property;
 }
+
+function getVersioningMutator(
+  context: TCGCContext,
+  service: Namespace,
+  apiVersion: string,
+): unsafe_MutatorWithNamespace {
+  const versionMutator = getVersioningMutators(context.program, service);
+  compilerAssert(
+    versionMutator !== undefined && versionMutator.kind !== "transient",
+    "Versioning service should not get undefined or transient versioning mutator",
+  );
+
+  const mutators = versionMutator.snapshots
+    .filter((snapshot) => apiVersion === snapshot.version.value)
+    .map((x) => x.mutator);
+  compilerAssert(mutators.length === 1, "One api version should not get multiple mutators");
+
+  return mutators[0];
+}
+
+export function handleVersioningMutationForGlobalNamespace(context: TCGCContext): Namespace {
+  const globalNamespace = context.program.getGlobalNamespaceType();
+  const service = listServices(context.program)[0];
+  if (!service) return globalNamespace;
+  const allApiVersions = getVersions(context.program, service.type)[1]
+    ?.getVersions()
+    .map((x) => x.value);
+  if (!allApiVersions) return globalNamespace;
```

Comment:
can we just return the `globalNamespace` for `all` version? the type graph without mutation seems should represent `all` just as previous we do with projection.

Reply:
oh I guess if you mean that all is the unmutated code then I think its fine too, no? what I think will be a nightmare right now at least is mutating every single version and try to merge that

Reply:
ooh ok, yeah let me give it a try then